core/commands/storage/session.go

```diff
@@ -875,13 +893,37 @@ func (ss *FileContracts) UpdateSessionStatus(status int, succeed bool, err error
 }
 
 func (ss *FileContracts) SendSessionStatusChan(status int, succeed bool, err error) {
+	ss.sendSessionStatusChan(status, succeed, err, false, context.TODO())
+}
+
+func (ss *FileContracts) SendSessionStatusChanPerMode(status int, succeed bool, err error, ctx context.Context) {
+	if ss.RunMode == OfflineSignMode {
+		ss.sendSessionStatusChan(status, succeed, err, true, ctx)
+	} else {
+		ss.sendSessionStatusChan(status, succeed, err, false, context.TODO())
```

Comment:
There shouldn't any more `context.Background()` or `context.TODO()`, those are prone to memory leaks, consider always using `NewGoContext(req.Context)` or something similar.